File: C/判断素数.c
#include<stdio.h>
#include<math.h>
#include<stdlib.h>
int ff(int x)
{
	int a,k;
	a=sqrt(x);
	if(x<2) return -1;
	for(k=2;k<=a;k++)
		if(x%k==0) return 0;
	return 1;
}
int main(int argc,char** argv)
{
	FILE *fp;
	if((fp=fopen("result.txt","w"))==NULL)
		return -1;
	int ret;
	ret=ff(atoi(argv[1]));
	if(ret==-1)
		fprintf(fp,"I");
	else if(ret==0)
		fprintf(fp,"N");
	else if(ret==1)
		fprintf(fp,"Y");
	fclose(fp);
	return 0;
}

File: Python/判断素数.py
import sys
import math
def ff(x):
	a=math.sqrt(x)
	k=2
	if x<2 :
		return -1
	else :
		while k<a:
			if x%k==0 :
				return 0
			k+=1
		return 1
x=ff(int(sys.argv[1]))
with open('result.txt','w') as fp:
	if x==-1 :
		fp.write('I')
	elif x==0 :
		fp.write('N')
	elif x==1 :
		fp.write('Y')